pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -617,12 +619,13 @@ func (r *reconcilerBase) setupOrTeardown(ctx context.Context, syncObj client.Obj
 	// Else - the object is being deleted.
 
 	if controllerutil.ContainsFinalizer(syncObj, metadata.ReconcilerFinalizer) {
-		// The object is being deleted, but the reconciler finalizer is still running.
-		// Wait for the reconciler finalizer to complete.
-		r.logger(ctx).Info("Waiting for Reconciler Finalizer to finish")
-		return nil
+		removed, err := r.handleReconcilerFinalizer(ctx, syncObj)
+		if !removed || err != nil {
+			return err
+		}
 	}
 
+	// The 'configsync.gke.io/reconciler' finalizer is removed successfully.
```

Comment:
This comment is slightly confusing because the `if` block in L621 and L628 look so similar. 
The reconciler finalizer is removed in the previous block, and in this block we check/handle the reconciler-manager finalizer.

Reply:
removed and added logs if not removed.